packages/phase/README.md

```diff
@@ -0,0 +1,134 @@
+# @loopback/phase
+
+This module defines `Phase` and `PhaseList` to organize a list of handlers by
+groups. It's rewritten in TypeScript based on
+https://github.com/strongloop/loopback-phase.
+
+## Installation
+
+```sh
+npm install --save @loopback/phase
+```
+
+## Basic Use
+
+### Handler
+
+A handler is a function that takes a `Context` object, an optional
+`HandlerChain` object and returns a `Promise`.
+
+When a handler is invoked as part of a chain, it can control how to proceed as
+follows:
+
+- Continue to invoke downstream handlers after it exits
+
+```ts
+async ctx => {
+  console.log(ctx.req.url);
+};
+```
+
+- Invoke downstream handlers within the method in cascading style
+
+```ts
+async (ctx, chain) => {
+  const start = process.hrtime();
+  await chain!.next();
```

Comment:
Please find a better type description for the callback function, one that does not require users to use `!` operator. If the framework always provide `chain` value, then this fact should be encoded in the type signature of the callback.